src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.VSCodeWorkspaces/WorkspacesHelper/VSCodeWorkspacesApi.cs

```diff
@@ -68,103 +68,131 @@ public List<VSCodeWorkspace> Workspaces
                                     {
-                                        string entries = sqlite_datareader.GetString(0);
-                                        if (!string.IsNullOrEmpty(entries))
-                                        {
-                                            VSCodeStorageEntries vscodeStorageEntries = JsonSerializer.Deserialize<VSCodeStorageEntries>(entries);
-                                            if (vscodeStorageEntries.Entries != null)
-                                            {
-                                                vscodeStorageEntries.Entries = vscodeStorageEntries.Entries.Where(x => x != null).ToList();
-                                                foreach (var entry in vscodeStorageEntries.Entries)
-                                                {
-                                                    bool isWorkspaceFile = false;
-                                                    var uri = entry.FolderUri;
-                                                    if (entry.Workspace != null && entry.Workspace.ConfigPath != null)
-                                                    {
-                                                        isWorkspaceFile = true;
-                                                        uri = entry.Workspace.ConfigPath;
-                                                    }
-
-                                                    var workspace = ParseVSCodeUri(uri, vscodeInstance, isWorkspaceFile);
-                                                    if (workspace != null)
-                                                    {
-                                                        results.Add(workspace);
-                                                    }
-                                                }
-                                            }
-                                        }
+                                        isWorkspaceFile = true;
+                                        uri = entry.Workspace.ConfigPath;
                                     }
-                                }
 
-                                sqliteConnection.Close();
+                                    var workspace = ParseVSCodeUri(uri, vscodeInstance, isWorkspaceFile);
+                                    if (workspace != null)
+                                    {
+                                        dbFileResults.Add(workspace);
+                                    }
+                                }
                             }
                         }
-                        catch (Exception ex)
-                        {
-                            var message = $"Failed to deserialize ${vscode_storage}";
-                            Log.Exception(message, ex, GetType());
-                        }
                     }
                 }
 
-                return results;
+                sqliteConnection.Close();
```

Comment:
why not put using statement?

Reply:
Thx. i missed that. I adjusted it to close the connection in the `finalize`